### PR TITLE
Exhibition related content

### LINKED ISSUE #1432: PP: build and release books module (stretch)
## Type

- ✨ Feature
- 🐛 Bug
- 🚑 Health report

## Needs

- [ ] User research
- [ ] Content from Content Team
- [ ] New visual design
- [ ] Analytics and/or component tracking

## Acceptance Criteria

- [ ] The interface should do this thing
- [ ] and this thing
- [ ] and this thing.

## Problem to be solved / User Story

In order to
 _(why? accomplish a goal/get a benefit)_

As a
 _(who? user/persona)_

I need/want/expect 
_(what? to do this thing)_

### PR DESCRIPTION
**Pull request by gestchild**
Closes #1432 #1469 #1470

## Type
✨ Feature  

## Value
Adds related content to exhibition page

## Screenshot
(still need to get final content and have design review)
screencapture-localhost-3000-exhibitions-wzwh4ioaaj3usf86-1505403106502

screencapture-localhost-3000-exhibitions-wzwh4ioaaj3usf86-1505403201412

### CODE REVIEW

File: client/scss/components/_book-promo.scss [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+.book-promo__title {
+  transition: color 400ms ease;
+
+  .book-promo:hover &,
+  .book-promo:focus & {
+    color: color('smoke');
+  }
+}
```
Comment (jamesgorrie) [inline client/scss/components/_book-promo.scss]: I might be being silly - but why is this all bespoke?
Could the grids not do what we needed?

File: server/services/exhibitions.js [hunk 2]
```diff
@@ -6,6 +6,16 @@ import {RichText} from 'prismic-dom';
...
 
+function exhibitionPromoToPromo(item) {
+  const promo = {
+    url: item.link.url,
+    title: item.title[0].text, // why is title an array?
+    description: item.description[0].text,
+    image: prismicImageToPicture(item)
+  };
+  return promo;
+};
```
Comment (jamesgorrie) [inline server/services/exhibitions.js]: Could we add the return type?

File: server/services/exhibitions.js [hunk 3]
```diff
@@ -28,6 +38,12 @@ export async function getExhibition(id: string, previewReq: ?Request): Promise<?
   const standfirst = bodyParts.find(p => p.type === 'standfirst');
   const imageGallery = bodyParts.find(p => p.type === 'imageGallery');
 
+  const relatedContent = exhibition.data.related;
```
Comment (jamesgorrie) [inline server/services/exhibitions.js]: As mentioned in the content model, I really think the word related is naming it something we wish it was, but isn't.

Would something like `promoLinks` make more sense?

File: server/views/components/book-promo/book-promo.njk [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{% set hasLink = model.url %}
+<div class="test">
```
Comment (jamesgorrie) [inline server/views/components/book-promo/book-promo.njk]: 👀

File: server/views/components/book-promo/book-promo.njk [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    {% endif %}
+    {% if model.subTitle %}
+      <h4 class="{{ {s:'HNM4'} | fontClasses }} {{ {s:0} | spacingClasses({margin: ['top', 'bottom']}) }}">{{model.subTitle}}</h4>
+    {% endif %}
+    {% if model.description %}
+       <p class="{{ {s:'HNL4'} | fontClasses }} {{ {s:2} | spacingClasses({margin: ['top']}) }} {{ {s:0} | spacingClasses({margin: ['bottom']}) }}">{{model.description}}</p>
+    {% endif %}
+  </div>
+</{{'a' if hasLink else 'div'}}>
+</div>
```
Comment (jamesgorrie) [inline server/views/components/book-promo/book-promo.njk]: Just a question around how we create components.

We've abstracted this into a container where there is nothing massively distinct about it - could it not just live directly in the template?

File: server/views/pages/exhibition.njk [hunk 6]
```diff
@@ -151,13 +150,67 @@
...
       </div>
     </div>
   {% endif %}
 
+  {% if exhibition.relatedGalleries.length > 0 %}
+    <div class="row {{ {s:5} | spacingClasses({margin: ['bottom']}) }}">
+      <div class="container">
+        <div class="grid">
+          <div class="{{ {s:12, m:12, l:12, xl:12} | gridClasses }}">
+            <h2 class="{{ {s:'WB4'} | fontClasses }}">Gallery highlights</h2>
```
Comment (jamesgorrie) [inline server/views/pages/exhibition.njk]: Is this really what we're calling it?

File: server/views/pages/exhibition.njk [hunk 7]
```diff
@@ -151,13 +150,67 @@
...
+            {% for promo in promoList %}
+              <div class="{{ gridSizes | gridClasses }}">
+                {% componentV2 promoComponent, promo,  {}, data %}
+              </div>
+            {% endfor %}
+          </div>
+        </div>
+      </div>
+    {% endif %}
+  {% endmacro %}
```
Comment (jamesgorrie) [inline server/views/pages/exhibition.njk]: Nice.

File: client/scss/components/_book-promo.scss (hunk 1 above)
Comment (gestchild) [reply client/scss/components/_book-promo.scss]: we use the grids for the layout of the promos as individual things, but the way the promo displays internally needs to be bespoke

File: server/services/exhibitions.js (hunk 2 above)
Comment (gestchild) [reply server/services/exhibitions.js]: yep

File: server/services/exhibitions.js (hunk 3 above)
Comment (gestchild) [reply server/services/exhibitions.js]: As discussed will make this relatedPromos

File: server/views/pages/exhibition.njk (hunk 6 above)
Comment (gestchild) [reply server/views/pages/exhibition.njk]: Think so, @Heesoomoon ?

File: server/services/exhibitions.js (hunk 3 above)
Comment (gestchild) [reply server/services/exhibitions.js]: Will do this in another PR after the demo

**Pull request merged**